hbase-server/src/test/java/org/apache/hadoop/hbase/client/TestMetaCache.java

```diff
@@ -459,6 +460,20 @@ public void testUserRegionLockThrowsException() throws IOException, InterruptedE
 
       assertTrue(client1.getException() instanceof LockTimeoutException
         ^ client2.getException() instanceof LockTimeoutException);
+
+      // obtain the client metrics
+      MetricsConnection metrics = conn.getConnectionMetrics();
+      long queueCount = metrics.getUserRegionLockQueue().getCount();
+      assertTrue(queueCount == 2);
+
+      long timeoutCount = metrics.getUserRegionLockTimeout().getCount();
+      assertTrue(timeoutCount == 1);
+
+      long waitingTimerCount = metrics.getUserRegionLockWaitingTimer().getCount();
+      assertTrue(waitingTimerCount == 1);
+
+      long heldTimerCount = metrics.getUserRegionLockHeldTimer().getCount();
```

Comment:
we could assert something basic about the timer, since the comments here suggest that we will hold the lock for 2 seconds. So we could assert that `metrics.getUserRegionLockHeldTimer().getSnapshot().getMax() >= 2000` milliseconds. (Probably better, just to avoid flaky tests, by subtracting some epsilon in case of time travel or something weird..., i.e. >=1500ms?)

Reply:
Yes, it is actually always a tiny greater than 2 seconds.

Reply:
I understand that it is taking around 2 seconds on your machine. Are we using some EnvironmentEdgeMananger tricks to ensure that it will take more than 2 seconds? If not then this will create flaky tests if we run the test in some powerful machine where it will take less than 2 seconds. @vli02